### PR TITLE
Clean-up of connection handler for better maintainability.

### PR DESCRIPTION
**Pull request by aentinger**

### CODE REVIEW

File: src/Notes-Lego [hunk 1]
```diff
@@ -0,0 +1,5 @@
+STM32
+MicroPython
+LittleFS (Wear Levelling)
+Motor APIs are non-blocking, callback on completion
+MicroPython allows cooperative multithreading which is used for this feature
```
Comment (per1234) [inline src/Notes-Lego]: Was it intended to commit this file?

**Comment by aentinger** [comment]
Successfully tested on PROD with
* `MKR 1000` :heavy_check_mark:
* `MKR WiFi 1010` :heavy_check_mark:
* `MKR GSM 1400` :heavy_check_mark:
* `MKR WAN 1310` :heavy_check_mark:
* `ESP8266` :heavy_check_mark:

@AlbyIanna can you please test with `MKR NB 1500` and `MKR WAN 1310`?

**Comment by AlbyIanna** [comment]
@Rocketct waiting for your tests

**Comment by Rocketct** [comment]
From Italy, with our test SIM the connection fall when the module try to open a socket to the broker,but this is expected due to the SIM's plan.

Today i plan another test in other country and i'll let you know the results as soon as i get it

**Comment by Rocketct** [comment]
test3ed form ourteam in malmo and wrks properly, the getting started and the cloud

**Pull request merged**